docs_src/design/adr/device-service/0012-DeviceService-Filters.md

```diff
@@ -0,0 +1,148 @@
+- the value was the same as some or not the same as some specified value or values (for strings, boolean and other non-numeric values)
+- the name (the device resource name which used to be the value descriptor) matched a particular value; in other words match `temperature` or `humidity` as example device resources.
+
+Unlike application services, there is not a need to filter on a device name (or identifier).  Simply disable the device in the device service if all `Event/Readings` are to be stopped for the device.
+
+In the case that all `Readings` of an `Event` are filtered, it is assumed the entire `Event` is deemed to be worthless and not sent to core data by the device service.  If only some `Readings` from and `Event` are filtered, the `Event` minus the filtered `Readings` would be sent to core data.
+
+### Design / Architecture
+
+A new function interface shall be defined that, when implemented, performs a Reading Filter operation.  A ReadingFilter function would take a parameter (an `Event` containing readings), check whether the `Readings` of the `Event` match on the filtering configuration (see below) and if they do then remove them from the `Event`.  The ReadingFilter function would return the `Event` object (minus filtered `Readings`) or `nil` if the `Event` held no more `Readings`.  Pseudo code for the generic function is provided below.
+
+``` go
+func (f Filter) ReadingFilter(lc logger.LoggingClient, event *models.Event) (*models.Event, error) {
+    // depending on impl; filtering for values in/out of a range, >, <, =, same, not same, from a particular name (device resource), etc.
+    if (len(event.Reading )) > 0)
+        return event
+    else
+        return nil
+}
+```
+
+Based on current needs/use cases, implementations of the function interface could include the following filter functions:
+
+``` go
+func (f Filter) FilterByValueInRange (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByValueOutRange (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByValueNotEqual (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByValueNotEqual (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+
+func (f Filter) FilterByResourceNamesMatch (lc logger.LoggingClient, event *models.Event) (*models.Event, error) {}
+```
+
+!!! Note
+    The app functions SDK comes with `FilterByDeviceName` and `FilterByResourceName` functions today. The FilterByResourceName would behave similarly to FilterByResourceNameMatch.
+
+    The Filter structure houses the configuration parameters for which the filter functions work and filter on.
+
+A NewFilter function creates, initializes and returns a new instance of the filter based on the configuration provided.
+
+``` go
+func NewReadingFilter(filterValues []string) Filter {
```

Comment:
I would change this to `NewReadingNameFilter(filterValues []string) Filter`  The Value filters will need different configuration.